src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -285,7 +285,7 @@ protected Scope ParentScope
             get
             {
                 Debug.Assert(this.scopeStack.Count > 1);
-                return this.scopeStack.Scopes.Skip(1).First();
+                return this.scopeStack.Parent;

```

Comment:
The `Parent` property is implemented this way:
```
            internal Scope Parent
            {
                get
                {
                    Debug.Assert(this.scopes.Count > 1, "this.scopes.Count > 1");
                    Scope current = this.scopes.Pop();
                    Scope parent = this.scopes.Peek();
                    this.scopes.Push(current);
                    return parent;
                }
            }
```
I'd be curious whether this is more performance than the use of `Skip(1)`...

Reply:
@henning-krause Someone actually raised this same question recently. Why we couldn't use a list. As a matter of fact, if you look at [the documentation](https://github.com/OData/odata.net/blob/b2d8bf7c5b3c3f90fd505e508888d528c0ae77b7/src/Microsoft.OData.Core/ODataWriterCore.cs#L3740) for the `scopes` field in the `ScopeStack` class, it talks about using a list to store the scopes. That is currently being looked at.

Reply:
> @henning-krause Someone actually raised this same question recently. Why we couldn't use a list. As a matter of fact, if you look at [the documentation](https://github.com/OData/odata.net/blob/b2d8bf7c5b3c3f90fd505e508888d528c0ae77b7/src/Microsoft.OData.Core/ODataWriterCore.cs#L3740) for the `scopes` field in the `ScopeStack` class, it talks about using a list to store the scopes. That is currently being looked at.

@henning-krause FYI: 4th bullet point on the PR description https://github.com/OData/odata.net/pull/2204

Reply:
How often do we call .Parent? Would it ever change? Would it be worth caching the fist time the user gets it?

Reply:
@mikepizzo I think the `.Parent` property is called a lot, and caching it would make sense, although I don't have done any performance testing as of yet. But since we have to think about cache invalidation as well I would leave that change for a future PR.